src/workerd/api/filesystem.c++

```diff
@@ -0,0 +1,1213 @@
+#include "filesystem.h"
+
+#include "blob.h"
+
+#include <workerd/api/streams/standard.h>
+#include <workerd/api/url-standard.h>
+#include <workerd/api/url.h>
+
+#include <kj/encoding.h>
+
+namespace workerd::api {
+
+namespace {
+constexpr kj::StringPtr getNameForFsType(FsType type) {
+  switch (type) {
+    case FsType::FILE:
+      return "file"_kj;
+    case FsType::DIRECTORY:
+      return "directory"_kj;
+    case FsType::SYMLINK:
+      return "symlink"_kj;
+    default:
+      KJ_UNREACHABLE;
+  }
+}
+
+constexpr FsType getTypeForName(kj::StringPtr name) {
+  if (name == "file"_kj) return FsType::FILE;
+  if (name == "directory"_kj) return FsType::DIRECTORY;
+  if (name == "symlink"_kj) return FsType::SYMLINK;
+  KJ_UNREACHABLE;
+}
+
+jsg::Url filePathToUrl(FilePath& path) {
```

Comment:
```suggestion
jsg::Url filePathToUrl(const FilePath& path) {
```

Reply:
-1 ... that causes problems a bit further down ... Specifically `FilePath` here is a `kj::OneOf`... if we make this a const ref, then the individual union types also become const refs... which causes a problem when we try to access the `legacyUrl` case since those methods aren't marked `const`

![image](https://github.com/user-attachments/assets/a1fd966f-17ae-406a-a74f-35edbb5fafbd)

I'd rather not have to go make changes to the `URL` class here so for now I'll keep this as a non-const reference and we can revisit later.